src/configs/passport.ts

```diff
@@ -2,9 +2,57 @@ import passport from 'passport'
 import { Strategy as JwtStrategy } from 'passport-jwt'
 import { dataSource } from './dbConfig'
 import Profile from '../entities/profile.entity'
-import { JWT_SECRET } from './envConfig'
+import {
+  JWT_SECRET,
+  GOOGLE_CLIENT_ID,
+  GOOGLE_REDIRECT_URL,
+  GOOGLE_CLIENT_SECRET
+} from './envConfig'
 import type { Request } from 'express'
 
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20'
+import { findOrCreateUser } from '../services/auth.service'
+import { type User } from '../types'
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: GOOGLE_CLIENT_ID,
+      clientSecret: GOOGLE_CLIENT_SECRET,
+      callbackURL: GOOGLE_REDIRECT_URL,
+      scope: ['profile', 'email'],
+      passReqToCallback: true // Add this line
+    },
+    async function (
+      req: Request, // Add this line
```

Comment:
```suggestion
      req: Request,
```
 Remove the comments in this section as well @mayura-andrew